src/SFA.DAS.OnBoardingTool.Application/Atlassian/Services/AtlassianService.cs

```diff
@@ -0,0 +1,227 @@
+using System.Net.Http;
+using System;
+using Microsoft.Extensions.Configuration;
+using System.Diagnostics;
+using SFA.DAS.OnBoardingTool.Infrastructure.Configuration;
+using Microsoft.Extensions.Options;
+using FluentValidation;
+using SFA.DAS.OnBoardingTool.Infrastructure.Validators;
+using SFA.DAS.OnBoardingTool.Infrastructure.Models;
+
+namespace SFA.DAS.OnBoardingTool.Application.Atlassian
+{
+    public class AtlassianService : IAtlassianService
+    {
+        private readonly IApiClient _apiClient;
+        private readonly ILogger<AtlassianService> _log;
+        private readonly OnBoardingToolFunctions _config;
+        private readonly AbstractValidator<Value> _validator;
+        
+        public AtlassianService(
+            IApiClient apiClient,
+            ILogger<AtlassianService> log,
+            IOptions<OnBoardingToolFunctions> config,
+            AbstractValidator<Value> validator)
+        {
+            _config = config.Value;
+            _apiClient = apiClient;
+            _log = log;         
+            _validator = validator;
+        }
+
+        public async Task ProcessCreateUserRequests()
+        {
+            _log.LogInformation(_config.ToString());
+            _log.LogInformation($"Processing create user requests");
+
+            var requests = await GetUserRequests();
+
+            if(requests.Size == 0)
+            {
+                _log.LogInformation($"No create user requests to process");
+                return;
+            }
+
+            var tasks = new List<Task>();
+
+            foreach (var request in requests.Values
+                .AsEnumerable()
+                .Where(r => r.Fields.Customfield_12105.Value.Equals("grant", StringComparison.InvariantCultureIgnoreCase)))
+            {                
+                var email = request.Fields.Customfield_12104;
+
+                _log.LogInformation($"Processing request for email {email} from reporter {request.Fields.Reporter.DisplayName}");
+
+                var result = _validator.Validate(request, opts => opts.IncludeRuleSets(ValueValidator.CanCreate));
+                if(!result.IsValid)
+                {
+                    _log.LogInformation($"Request failed validation {string.Join(',', result.Errors.Select(err => err.ErrorMessage).ToArray())}");
+                    continue;
+                }
+                    
+                tasks.Add(CreateUser(request, new User { Username = email,  Email = email })
+                .ContinueWith(TransitionRequest, TaskContinuationOptions.OnlyOnRanToCompletion));
+            }
+
+            Task.WaitAll(tasks.ToArray());
+
+            _log.LogInformation($"Finished processing create requests");
+        }
+
+        public async Task ProcessRevokeUserRequests()
+        {
+            _log.LogInformation(_config.ToString());
+            _log.LogInformation($"Processing revoke user requests");
+
+            var requests = await GetUserRequests();
```

Comment:
I'm guessing the Api doesn't have any query parameters to only find the requests that have a custom field value of "revoke" or "grant" to remove the Where restriction on the loop?

Reply:
Correct there's no way (currently) to filter that at the API level